### PR TITLE
Test code proposed refactor - used create wallet tests as an example

### PR DESCRIPTION
**Pull request by dalebennett1992**
Having looked at our tests I thought it might be worth considering alternate approaches for writing our test code while the framework is still in its infancy. I think these changes will result in more maintainable test code and reusable methods that should make creating more tests faster/easier. Would be interested to get thoughts on it as an approach going forward.

I have targeted just a couple of tests for now.

### CODE REVIEW

**Comment by dalebennett1992** [comment]
> I like this a lot, seem simpler, but I'd be interested to hear @radoslaw-szpiech thoughts on this as he has done much more of this than I have 👍

nice, @radoslaw-szpiech if you are happy with this approach we could get this down? Let me know

**Comment by radoslaw-szpiech** [comment]
@dalebennett1992 @dexturr I remember the discussion that we were having at the beginning of cypress testing, that we want to get rid of the page object model approach, and put everything directly into test files, to make it easier for debug, and have clear picture what is happening in test. I think we can create some common functions and reuse that, to avoid lots of repetition, but I would rather not use full page object model - no point of creating separate function for a `click` action on specific element, that is used only once or twice ;) Just use  common sense doing that :)

**Comment by radoslaw-szpiech** [comment]
if we have situation like this - 
image
that means we did it wrong :D

File: frontend/automation/support/pages/create-wallet.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { expect, Locator, Page } from '@playwright/test'
+
+export class CreateWallet {
```
[the remark this answers is not in the document]
Comment (dalebennett1992) [reply frontend/automation/support/pages/create-wallet.ts:3]: > no need to create class, as we won't create different instances of it.

So you guys have evidently had a discussion about the appropriateness of the page object model and I am late to the party. Is there anything I can go and read? Because as far as I am concerned it seems very applicable here. 

I have a few reasons for feeling tests should be written this way. In my experience any tests written in the currently existing fashion will not be particularly maintainable given that there is functionality hand cranked in multiple places along with repeated locators, not to mention it is a little tough for me (especially as I am new) to understand what is going on. Also as you mentioned about the code increase- more code != more maintenance in this instance. If it is a framework we want to upscale then I think this makes it easier to scale out and maintain since there is a single source of truth. Would be good to have a discussion so I can understand the team's approach better :) @ValentinTrinque @dexturr @radoslaw-szpiech

[the remark this answers is not in the document]
Comment (radoslaw-szpiech) [reply frontend/automation/support/pages/create-wallet.ts:3]: I think I was misunderstood a little :D Let me rephrase that.
The talk we had was a year ago, I don't think we have any archive of that on Slack. But... it's history :D We don't need to strictly keep to what we said a year ago, also we don't need to strictly keep to any approaches, e.g. POM).
The page objects you created are fine, there's a lot of reusability here. They can be used in other tests files as well, so GJ! What I said about not creating unnecessary functions was more like general advice for the future to not fall into trap of creating separate additional functions, that are handling some simple action, and they are not re-used everywhere. 
The thing about class - no need for using class, as we probably won't create different instances of it, with different parameters or keeping it's state, so it just adds more code and no real value. I will create a commit to show what I meant :)

[the remark this answers is not in the document]
Comment (radoslaw-szpiech) [reply frontend/automation/support/pages/create-wallet.ts:3]: @dalebennett1992 https://github.com/vegaprotocol/vegawallet-desktop/pull/616 take a look here

[the remark this answers is not in the document]
Comment (radoslaw-szpiech) [reply frontend/automation/support/pages/create-wallet.ts]: Merged. :) Classes are just more than what we really need here

**Comment by radoslaw-szpiech** [comment]
@dalebennett1992 Can you reuse the functions you created in other tests as well?

**Comment by dalebennett1992** [comment]
> @dalebennett1992 Can you reuse the functions you created in other tests as well?

theoretically you can use these functions anywhere really, as many tests as you like..

File: frontend/automation/support/pages/create-wallet.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dalebennett1992) [reply frontend/automation/support/pages/create-wallet.ts]: yeah I left a comment, essentially it is achieving the same thing tbh but if that is what you prefer then I am pretty easy about it. Thanks

**Comment by dalebennett1992** [comment]
so I have updated wallet.test.ts to use the new functionality. I think anything remaining should be incremental so the scope of this PR doesn't shift too much or get too complex to review. I will target other areas (perhaps on a per test file basis) in between other bits of work.

**Review by radoslaw-szpiech — APPROVED** [review]

**Review by radoslaw-szpiech — APPROVED** [review]
